lib/phoenix_token_auth/registration_flow.ex

```diff
@@ -0,0 +1,28 @@
+defmodule PhoenixTokenAuth.RegistrationFlow do
+  alias PhoenixTokenAuth.Mailer
+  alias PhoenixTokenAuth.Util
+
+  def user_created_by_username?(changeset) do
+    if changeset.valid? do
+      Util.repo.transaction(fn ->
+        case Util.repo.insert(changeset) do
+          {:ok, _} -> true
+          {:error, _} -> false
+        end
+      end)
+    end
+  end
+
+  def user_created_and_emailed?(changeset, confirmation_token, conn) do
+    if changeset.valid? do
+      Util.repo.transaction(fn ->
```

Comment:
So my thinking with keeping the `valid?` check is that we wouldn't need to hit the DB to check constraints if we already know the record is invalid, because at that point we already know the insert will fail. But it may read better to simply attempt to insert the record.